src/interface/file.rs

```diff
@@ -53,84 +31,45 @@ pub fn removefile(filename: String) -> std::io::Result<()> {
-
-    if filename.len() > MAX_FILENAME_LENGTH {
-        panic!("ArgumentError: Filename exceeds maximum length.")
-    }
-
-    if !filename.chars().all(is_allowed_char) {
-        println!("'{}'", filename);
-        panic!("ArgumentError: Filename has disallowed characters.")
-    }
-
-    match filename.as_str() {
-        "" | "." | ".." => panic!("ArgumentError: Illegal filename."),
-        _ => {}
-    }
-
-    if filename.starts_with(".") {
-        panic!("ArgumentError: Filename cannot start with a period.")
-    }
+pub fn openfile(filename: String, filesize: usize) -> std::io::Result<EmulatedFile> {
+    EmulatedFile::new(filename, filesize)
 }
 
-pub fn openfile(filename: String, create: bool) -> std::io::Result<EmulatedFile> {
-    EmulatedFile::new(filename, create)
+pub fn openmetadata(filename: String) -> std::io::Result<EmulatedFile> {
+    EmulatedFile::new_metadata(filename)
 }
 
 #[derive(Debug)]
 pub struct EmulatedFile {
     filename: String,
-    abs_filename: RustPathBuf,
     fobj: Option<Arc<Mutex<File>>>,
     filesize: usize,
 }
 
 pub fn pathexists(filename: String) -> bool {
-    assert_is_allowed_filename(&filename);
     let path: RustPathBuf = [".".to_string(), filename.clone()].iter().collect();
     path.exists()
 }
 
 impl EmulatedFile {
 
-    fn new(filename: String, create: bool) -> std::io::Result<EmulatedFile> {
-        assert_is_allowed_filename(&filename);
-
-        if OPEN_FILES.contains(&filename) {
-            panic!("FileInUse");
-        }
-
-        let path: RustPathBuf = [".".to_string(), filename.clone()].iter().collect();
+    fn new(filename: String, filesize: usize) -> std::io::Result<EmulatedFile> {
 
-        let f = if !path.exists() {
-            if !create {
-              panic!("Cannot open non-existent file {}", filename);
-            }
+        let f = OpenOptions::new().read(true).write(true).create(true).open(filename.clone()).unwrap();
+        Ok(EmulatedFile {filename: filename, fobj: Some(Arc::new(Mutex::new(f))), filesize: filesize})
+    }
 
-            OpenOptions::new().read(true).write(true).create(true).open(filename.clone())
-        } else {
-            OpenOptions::new().read(true).write(true).open(filename.clone())
-        }?;
+    fn new_metadata(filename: String) -> std::io::Result<EmulatedFile> {
 
-        let absolute_filename = fs::canonicalize(&path)?;
+        let f = OpenOptions::new().read(true).write(true).create(true).open(filename.clone()).unwrap();
 
-        OPEN_FILES.insert(filename.clone());
         let filesize = f.metadata()?.len();
 
-        Ok(EmulatedFile {filename: filename, abs_filename: absolute_filename, fobj: Some(Arc::new(Mutex::new(f))), filesize: filesize as usize})
-
+        Ok(EmulatedFile {filename: filename, fobj: Some(Arc::new(Mutex::new(f))), filesize: filesize as usize})
     }
 
+
     pub fn close(&self) -> std::io::Result<()> {
-        OPEN_FILES.remove(&self.filename);
         Ok(())
```

Comment:
Our intention in retaining these is to avoid extensive changes to the logic in fs_calls.rs at this stage, correct?